### PR TITLE
[D1] teach init --from-dash about d1

### LINKED ISSUE #6253: 🐛 BUG: init --from-dash doesn't know about D1
### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version(s) of the tool(s) are you using?

latest

### What version of Node are you using?

latest

### What operating system and version are you using?

Mac

### Describe the Bug

.

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

### PR DESCRIPTION
**Pull request by rozenmd**
## What this PR solves / how to test
Run `wrangler init --from-dash <my-d1-worker-here>`, check if the wrangler.toml contains unsafe bindings

Fixes #6253

## Author has addressed the following

- Tests
  - [x] TODO (before merge)
  - [ ] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [x] I don't know
  - [ ] Required / Maybe required
  - [ ] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: bug fix

### CODE REVIEW

File: packages/wrangler/src/init.ts [hunk 1]
```diff
@@ -1035,6 +1052,18 @@ export function mapBindings(bindings: WorkerMetadataBinding[]): RawConfig {
...
 						}
 						break;
+					case "d1":
+						{
+							configObj.d1_databases = [
+								...(configObj.d1_databases ?? []),
+								{
+									binding: binding.name,
+									database_id: binding.id,
+									database_name: d1BindingsWithInfo[binding.id].name,
```
Comment (rozenmd) [inline packages/wrangler/src/init.ts:1062]: You might be tempted to ask why we don't just teach EWC about `database_name` - the thing is, EWC doesn't store or care about the name of the DB, only D1's API knows it, so we need to fetch it if we want Workers initialised with `--from-dash` to be capable of running `wrangler d1` commands.

**Review by penalosa — COMMENTED** [review]
Looks mostly good, but could you add a test in `init.test.ts` (or expand an existing test to include D1)?

**Review by threepointone — APPROVED** [review]
the code looks fine, stamping. unfortunate that this will make projects with d1 bindings a little slower to pull down, but that's not something we can fix here anyway.

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
Just running e2e tests in case this breaks one of them.

**Pull request merged**